funcbench/README.md

```diff
@@ -45,53 +50,67 @@ Errors out if there are no sub-benchmarks.
 
-#### Example Github actions workflow file to pass in --input flag.
+- Create GitHub actions workflow file (see below) that is executed when an `repository_dispatch` event is on.
+- Also add GKE auth info (PROJECT_ID / ZONE / CLUSTER_NAME) as secrets
+- Read LAST_COMMIT_SHA / BRANCH / PR_NUMBER / REGEX from event payload into environment variables
+- Ignore this if your benchmark lasts less than 60 minutes.
+  - Provide a GitHub token as a secret, here named `PERSONAL_TOKEN`, to avoid conflicts with default `GITHUB_TOKEN`.
 
-> TODO: No longer using `issue_comment`, to be replaced with commentMonitor usage.
+#### Example GitHub action workflow file
 
-```
-on: issue_comment // Workflow is executed when a pull request comment is created.
-name: Benchmark
+```yaml
+on: repository_dispatch
+name: Funcbench Workflow
 jobs:
-  commentMonitor:
+  run_funcbench:
+    name: Running funcbench
+    if: github.event.action == 'funcbench_start'
     runs-on: ubuntu-latest
+    env:
+      AUTH_FILE: ${{ secrets.PROMBENCH_GKE_AUTH }}
+      CLUSTER_NAME: ${{ secrets.CLUSTER_NAME }}
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      LAST_COMMIT_SHA: ${{ github.event.client_payload.LAST_COMMIT_SHA }}
+      PROJECT_ID: ${{ secrets.PROJECT_ID }}
+      PR_NUMBER: ${{ github.event.client_payload.PR_NUMBER }}
+      TEST_INFRA_REPO: https://github.com/prometheus/test-infra.git # put your repo here if test your own test-infra
+      ZONE: ${{ secrets.ZONE }}
     steps:
-    - name: commentMonitor
-      uses: docker://prominfra/comment-monitor:latest
-      env:
-        COMMENT_TEMPLATE: 'The benchmark has started.' // Body of a comment that is created to announce start of a benchmark.
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} // Github secret token/
+    - name: Prepare nodepool
+      uses: docker://prominfra/prombench:latest
       with:
-        args: '"^/funcbench ?(?P<BRANCH>[^ B\.]+)? ?(?P<REGEX>\.|Bench.*|[^ ]+)?'
-    - name: benchmark
-      uses: docker://prominfra/funcbench:latest
+        args: make funcbench_nodepool_create
+    - name: Run funcbench
+      uses: docker://prominfra/prombench:latest
       env:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} // Github secret token/
-```
-
-#### Set up
-
-This tool is meant to be used with a Github action. The action itself is, to a large degree, unusable alone, as you need
-to combine it with another Github action that will provide necessary files to it. At this time, the only action it is
-supposed to work with, is [comment-monitor](https://github.com/prometheus/prombench/tree/master/tools/commentMonitor).
+        BRANCH: ${{ github.event.client_payload.BRANCH }}
+        GITHUB_ORG: prominfra
+        GITHUB_REPO: prometheus
+        REGEX: ${{ github.event.client_payload.REGEX }}
+      with:
+        args: make funcbench_resource_apply
```

Comment:
Would this not need deleting the funcbench namespace or deleting the nodepool afterwards?

Reply:
Added more steps below to recycle them.